### ISSUE TITLE
UI Components always bundle storage, even if it is not used

### ISSUE THREAD

**Issue by atyshka** [aws-amplify/amplify-js#7484]
**Describe the bug**
Any use of the ui components necessitates the inclusion of the storage api, which takes up a whole megabyte. For those of us who do not use storage components but only things like auth components, this is a massive waste. 20% of my app bundle is now this storage api which I am not using.

**To Reproduce**

Import either the whole component library via `import "@aws-amplify/ui-vue";` or modularly by 
```
import {
  AmplifyAuthenticator,
  AmplifySignIn,
} from "@aws-amplify/ui-vue";
```

**Expected behavior**
The modular import, including only the authenticator and sign in, would not bundle the s3 storage dependency.

**Code Snippet**
Please provide a code snippet or a link to sample code of the issue you are experiencing to help us reproduce the issue. (Be sure to remove any sensitive data)
```
import {
  AmplifyAuthenticator,
  AmplifySignIn,
} from "@aws-amplify/ui-vue";
```

**Screenshots**

![image](https://user-images.githubusercontent.com/19317207/103558693-09fa1a00-4e83-11eb-9e62-28566cac63ba.png)
The blue block on the right is the unnecessary inclusion of s3, comprising ~20% of the bundle size

**Comment by ericclemmons** [comment]
@atyshka You're absolutely right. Normally named imports like you gave above would tree-shake out unused modules. But the `@aws-amplify/ui-components` are built by StencilJS, resulting in a single bundle.

It'll take some research to see if we can make it tree-shakeable in a backwards compatible way.  One alternative is to use [`dist-custom-elements-bundle`](https://stenciljs.com/docs/custom-elements), which should result in single bundles per-component, but has its own trade-offs.  @wlee221 Have you looked into this as wel?

**Comment by wlee221** [comment]
I have not, but I should! This is related to discussions we had for #6802 as well. I was under the impression that bundles generated with `dist` should be tree shakable based on their [docs](https://stenciljs.com/docs/distribution). This is definitely worth investigating. @atyshka, can you share your `package.json`?

**Comment by wlee221** [comment]
On the other note, named imports aren't available `ui-vue`. You just need `import "@aws-amplify/ui-vue"` which would define the components ([docs](https://docs.amplify.aws/ui/auth/authenticator/q/framework/vue#installation)). Your code should complain `export 'AmplifyAuthenticator' was not found in '@aws-amplify/ui-vue'`, right?

**Comment by atyshka** [comment]
@wlee221 No, actually for some reason I do not; I see no errors in the compiler output nor the runtime console and the components appear correctly on screen.

**Comment by jamessouth** [comment]
Here's my bundle analysis from a React project. This is after going from [this, in the React docs:](https://docs.amplify.aws/start/getting-started/setup/q/integration/react#install-amplify-libraries)
```javascript
import Amplify from "aws-amplify";
import awsExports from "./aws-exports";
Amplify.configure(awsExports);
```
to [this, adapted from the JS docs:](https://docs.amplify.aws/start/getting-started/data-model/q/integration/js#testing-your-api)
```javascript
import Amplify from "@aws-amplify/auth";
import awsExports from "../aws-exports";
Amplify.configure(awsExports);
```
which seems to work the same but reduced the size of the dev mode vendor bundle from 31 to 11 megs! Prod mode from 1.2 to 1.02, parsed size! 
![Screenshot 2021-01-09 231800](https://user-images.githubusercontent.com/25541381/104115373-7edeb100-52d4-11eb-92d0-8d09b2118a2b.png)
The bundle includes apparently extraneous modules like client-s3, client-lex-runtime, amplify-chatbot-entry, and 7 instances of bn.js.

**Comment by atyshka** [comment]
@jamessouth Yes modular imports improve things in your case, but this issue is unrelated. This refers specifically to modular imports for UI components from StencilJS. As you can see, modular imports work just fine for other modules such as authentication.

**Comment by ericclemmons** [comment]
We'll be testing out 729657353212d408a415301bdd0a802937c4ec04 & e8e0c8421ab1085d7615350e91e70db670b45f25 to see if that helps resolve usage with `@aws-amplify/ui-components` directly 🙏

**Comment by atyshka** [comment]
@ericclemmons running latest release and seems to still be an issue, can you reproduce as well?

**Comment by atyshka** [comment]
It seems that e8e0c84 was merged but 7296573 was not?
